public/content/posts/performance-improvement-in-a-next.js-page/index.md

```diff
@@ -0,0 +1,120 @@
+
+![LCP score margin](/lcp.png "LCP Score")
+
+### Identify LCP of a page
+
+There are few ways we can identify the LCP of a page. One of easy was is to use latest Performance Insights tab in google chrome. Once you clicked "Measure page load" button it will run for a few seconds and then it will show some details related to performance. On the right side of the tab, you will be able to see web vitals and once you hover LCP it will highlight the LCP content of the page.
+
+![Google Chrome Performance Insights](/perf-insight.png "Google Chrome Performance Insights")
+
+### improve performance of the image
+
+Once we identify the LCP element of the page, we can try to make sure it loads fast relatively. Since in this case it's an image we can should load the image in "eager" mode instead of the "lazy" mode.
+
+```html
+<img src="imageUrl" loading="eager" />
+```
+
+since the eager is the default value for loading prop, in this case we can ignore the loading prop as well.
+
+[This Article](/posts/improving-images-on-fishbrain.com) from our blog also explains how we have improved the overall performance of the images.
+
+### Preload the LCP image
+
+Perloading the image inside next/head is another effective way to improve the LCP score.
+
+```jsx
+<Head>
+  <link rel="preload" href={imageUrl} as="image" />
+</Head>
+```
+
+This will ask the browser to load image upfront even before it discovers the image element.
+
+For testing a single page (url) performance, [web.dev/measure](https://web.dev/measure) is a great tool. After measuring the performance with this tool, it provides us some useful methods on how to improve our web vitals scores.
+
+### Next.js Dynamic imports
+
+One of the method it suggest is to reduce the initial response time. For this we could use [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import). It's a handy way to load components and libraries only when those are needed. Therefore we could use the dynamic imports when we are loading components conditionaly. This will also slightly reduce the build size.
+
+```jsx
+const DynamicCatchesSection = dynamic(async () =>
+  import("./components/CatchesSection")
+);
+```
+
+```jsx
+<DynamicCatchesSection
+  title="This is title"
+  regionCode={regionCode}
+  slug={slug}
+  countryCode={countryCode}
+/>
+```
+
+### Results
+
+Before:
+![Lighthouse Performance Desktop Before](/performance.png "Lighthouse Performance Desktop Before")
+
+After:
+![Lighthouse Performance Desktop After](/performance-2.png "Lighthouse Performance Desktop After")
+As you can see performance is quite good and specially the LCP score has improved a lot compared to the earlier version.
+
+**Note:** _The performance is tested in lighthouse tool for Desktop type. Score can be different from time to time and for mobile devices this score will be considerably lower since the mobile has less resource power than the Desktop devices._
```

Comment:
```suggestion
**Note:** _The performance is tested in Lighthouse using Desktop mode. Scores can be different from time to time and for mobile devices this score will be considerably lower since the mobile has fewer resources than Desktop devices._
```